components/announcement_bar/cloud_trial_announcement_bar/cloud_trial_announcement_bar.tsx

```diff
@@ -0,0 +1,193 @@
+        } else if (daysLeftOnTrial <= TrialPeriodDays.TRIAL_3_DAYS && daysLeftOnTrial > TrialPeriodDays.TRIAL_0_DAYS) {
+            dismissValue = '3_days_banner';
+        }
+        trackEvent(
+            TELEMETRY_CATEGORIES.CLOUD_ADMIN,
+            'click_close_banner_trial_period',
+        );
+        await this.props.actions.savePreferences(this.props.currentUser.id, [{
+            category: Preferences.CLOUD_TRIAL_BANNER,
+            user_id: this.props.currentUser.id,
+            name: CloudBanners.TRIAL,
+            value: `${dismissValue}`,
+        }]);
+    }
+
+    shouldShowBanner = () => {
+        const {isFreeTier} = this.props;
+        return isFreeTier;
+    }
+
+    isDismissable = () => {
+        const {daysLeftOnTrial} = this.props;
+        let dismissable = true;
+
+        if (daysLeftOnTrial < TrialPeriodDays.TRIAL_1_DAY) {
+            dismissable = false;
+        }
+        return dismissable;
+    }
+
+    showModal = () => {
+        if (this.isDismissable()) {
+            trackEvent(
+                TELEMETRY_CATEGORIES.CLOUD_ADMIN,
+                'click_upgrade_banner_trial_limit_not_reached',
+            );
+        } else {
+            trackEvent(
+                TELEMETRY_CATEGORIES.CLOUD_ADMIN,
+                'click_upgrade_banner_trial_limit_exceeded',
+            );
+        }
+        this.props.actions.openModal({
+            modalId: ModalIdentifiers.CLOUD_PURCHASE,
+            dialogType: PurchaseModal,
+        });
+    }
+
+    render() {
+        const {daysLeftOnTrial, preferences} = this.props;
+
+        if (isEmpty(this.props.analytics)) {
+            // If the analytics aren't yet loaded, return null to avoid a flash of the banner
+            return null;
+        }
+
+        if (!this.shouldShowBanner()) {
+            return null;
+        }
+
+        if ((preferences.some((pref) => pref.name === CloudBanners.TRIAL && pref.value === '14_days_banner') && daysLeftOnTrial > TrialPeriodDays.TRIAL_3_DAYS) ||
+            ((daysLeftOnTrial <= TrialPeriodDays.TRIAL_3_DAYS && daysLeftOnTrial > TrialPeriodDays.TRIAL_0_DAYS) && preferences.some((pref) => pref.name === CloudBanners.TRIAL && pref.value === '3_days_banner'))) {
+            return null;
+        }
+
+        const trialMoreThan3DaysMsg = (
+            <FormattedMessage
+                id='admin.billing.subscription.cloudTrial.moreThan3Days'
+                defaultMessage='Your trial has started! There are {daysLeftOnTrial} days left'
+                values={{daysLeftOnTrial}}
+            />
+        );
+
+        let bannerMessage;
+        switch (daysLeftOnTrial) {
+        case TrialPeriodDays.TRIAL_3_DAYS:
+            bannerMessage = t('admin.billing.subscription.cloudTrial.3daysLeftOnTrial');
+            break;
+        case TrialPeriodDays.TRIAL_2_DAYS:
+            bannerMessage = t('admin.billing.subscription.cloudTrial.2daysLeftOnTrial');
+            break;
```

Comment:
Should this be "You have {daysLeft} on trial" where it dinamically builds the message based on the amount of days left?